editor.planx.uk/src/routes/flow.tsx

```diff
@@ -86,58 +96,69 @@ const newNode = route(async (req) => {
   };
 });
 
-const editNode = route(async (req) => {
-  const { id, before = undefined, parent = undefined } = req.params;
-
-  const node = useStore.getState().getNode(id) as {
-    type: TYPES;
-    [key: string]: any;
-  };
-
-  const extraProps = {} as any;
-
-  if (node.type === TYPES.ExternalPortal)
-    extraProps.flows = await getExternalPortals();
+// If nodeId is invalid (e.g. a deleted node), fall back to flow
+const validateEdit = (route: Matcher<object, object>) =>
+  map((req) => {
+    const { team, flow, id } = req.params;
+    const node = useStore.getState().getNode(id);
+    if (!node) return redirect(`/${team}/${flow}`);
+    return route;
+  });
 
-  const type = SLUGS[node.type];
-
-  if (type === "checklist" || type === "question") {
-    const childNodes = useStore.getState().childNodesOf(id);
-    if (node.data?.categories) {
-      extraProps.groupedOptions = mapAccum(
-        (index: number, category: { title: string; count: number }) => [
-          index + category.count,
-          {
-            title: category.title,
-            children: childNodes.slice(index, index + category.count),
-          },
-        ],
-        0,
-        node.data.categories
-      )[1];
-    } else {
-      extraProps.options = childNodes;
+const editNode = validateEdit(
```

Comment:
The diff here is not super helpful - the only change to this function is that it's now wrapped in `validateEdit()`